### ISSUE TITLE
'DummyOptim' object has no attribute 'step'

### ISSUE THREAD

**Issue by baizh0u** [bghira/SimpleTuner#421]
When try to use deepspeed with multi-GPU to train sdxl, it will occurred 'DummyOptim' object has no attribute 'step' problem, Is there anyway to fixed that?

**Comment by bghira** [comment]
need more info. how are you launching the trainer? what is your accelerate config looking like?

**Comment by baizh0u** [comment]
> need more info. how are you launching the trainer? what is your accelerate config looking like?

Find you guys updated the code and also fixed this issue...because the lr_scheduler has been initiated twice...

**Comment by bghira** [comment]
it works now? the lr scheduler init still happens twice in the current master branch

**Comment by baizh0u** [comment]
I just commented out the code in line 8xx, forgot the exact line, it is also refer to initial the lr scheduler…I don’t think it is necessary when using deepspeed in training. This issue will not occur, if you do not use deepspeed for training.Bagheera ***@***.***>於2024年5月30日 20:53寫道：﻿
it works now? the lr scheduler init still happens twice in the current master branch

**Comment by baizh0u** [comment]
﻿I just commented out the code in line 8xx, forgot the exact line, it is also refer to initial the lr scheduler…I don’t think it is necessary when using deepspeed in training. This issue will not occur, if you do not use deepspeed for training.Bagheera ***@***.***>於2024年5月30日 20:53寫道：﻿
it works now? the lr scheduler init still happens twice in the current master branch

**Comment by bghira** [comment]
the problem is that the resume step count is typically not set at the start. would you like to open a PR where it doesn't run that line if `use_deepspeed_scheduler` isn't set?

**Comment by baizh0u** [comment]
Great to see!!!  I just have time to read the message!
